melodies_monet/driver.py

```diff
@@ -734,6 +734,42 @@ def plotting(self):
                         else:
                             pairdf = pairdf_all.reset_index().dropna(subset=[modvar])
 
+                        #JianHe: Determine if calcuate regulatory values
+                        if 'regulatory' in obs_plot_dict.keys():
+                            cal_reg = obs_plot_dict['regulatory']
+                        else:
+                            cal_reg = False
+                        #print('Paired dataframe without regulatory: ', cal_reg, modvar, p_label, pairdf)
+
+                        if cal_reg:
+                            df1 = pairdf.copy()
+                            df1.index = df1.time_local
+                            df2 = df1.groupby("siteid").resample('H').mean().reset_index()        
+                            
+                            if obsvar == 'PM2.5':  
+                                pairdf_reg = splots.make_24hr_regulatory(df2,[obsvar,modvar]).rename(index=str,columns={obsvar+'_y':obsvar+'_reg',modvar+'_y':modvar+'_reg'})
+                            elif obsvar == 'OZONE':
+                                pairdf_reg = splots.make_8hr_regulatory(df2,[obsvar,modvar]).rename(index=str,columns={obsvar+'_y':obsvar+'_reg',modvar+'_y':modvar+'_reg'})
+                            else:
+                                print('Warning: no regulatory caculations found for ' + obsvar) 
+
+                            if len(pairdf_reg[obsvar+'_reg']) == 0:
+                                print('No valid data for '+obsvar+'_reg')
+                                cal_reg = False
+                            else:
+                                #print('Paired dataframe with regulatory: ', obsvar, pairdf_reg)
+                                outname = "{}.{}.{}.{}.{}.{}.{}".format(grp, plot_type, obsvar+'_reg', startdatename, enddatename, domain_type, domain_name)
+                                use_ylabel = obs_plot_dict['ylabel2_plot']
```

Comment:
See above, can we change the name to be more specific like ylabel_reg_plot. Also can we change the code to be similar to that above for the ylabel_plot where if it is not provided you set use_ylabel = none. That way if people do not want to provide a name in the yaml file that's okay it will default to the name of the obsvar name.